questionpy_server/repository/models.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from datetime import datetime
+from functools import total_ordering
+from pathlib import Path
+
+from pydantic import BaseModel, PositiveInt
+from questionpy_server.utils.manfiest import ComparableManifest, SemVer
+
+
+class RepoMeta(BaseModel):
+    timestamp: datetime
+    size: PositiveInt
+    sha256: str
+
+
+@total_ordering
+class RepoPackageVersion(BaseModel):
+    version: SemVer
+    api_version: str
+
+    path: Path
+    size: PositiveInt
+    sha256: str
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, RepoPackageVersion):
+            return NotImplemented
+        return self.version < other.version
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, RepoPackageVersion):
+            return NotImplemented
+        return self.version == other.version
+
+
+class RepoPackageVersions(BaseModel):
+    manifest: ComparableManifest
+    versions: list[RepoPackageVersion]
+
+
+class RepoPackageIndex(BaseModel):
+    packages: list[RepoPackageVersions]
```

Comment:
Ich hatte mir hier noch eine Versionsnummer für das Dateiformat gewünscht. Vielleicht `package_index_version: int`?

Reply:
Im Zuge dessen können die Pakete des Index wieder direkt als Liste in der `PACKAGES.json.gz` aufgeführt werden (und nicht erst als Value von `packages`), da alle zukünftigen Meta-Daten in der `META.json` sein sollten.

Reply:
Es schadet aber auch nicht und gleichzeitig sind wir flexibel, falls wir doch mal Anpassungen vornehmen wollen.

Reply:
Das stimmt natürlich!